src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -454,6 +460,15 @@ def find_or_create_tag(self, tag, root_name=None, **new_symbol_args):
         except KeyError:
             if not root_name:
                 root_name = tag
+            if exact_name:
+                symbols = self.get_symbols()
+                existing_names = set(symbols.keys())
+                if root_name in existing_names:
```

Comment:
Maybe in a one-liner "if root_name in self.get_symbols().keys()" will be clearer.